src/snowflake/snowpark/session.py

```diff
@@ -577,6 +580,12 @@ def __init__(
             self._conn._get_client_side_session_parameter(
                 _PYTHON_SNOWPARK_AUTO_CLEAN_UP_TEMP_TABLE_ENABLED, False
             )
+            and pkg_resources.parse_version(self.version)
+            >= pkg_resources.parse_version(
```

Comment:
@sfc-gh-jdu can you add a unit test to test test with different self.version, and the configrued version number to make sure things works correctly

Reply:
added